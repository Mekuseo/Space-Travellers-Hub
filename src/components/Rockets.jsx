import React from 'react';
import { useSelector } from 'react-redux';

function Rockets() {
  const list = useSelector((state) => state.Rocket.rockets);
  return (
    <div className="mii">
      <ul>
        {list.map((lee) => (
          <li key={Math.floor(Math.random() * 1000)}>{lee}</li>
        ))}
      </ul>
    </div>
  );
}

export default Rockets;
