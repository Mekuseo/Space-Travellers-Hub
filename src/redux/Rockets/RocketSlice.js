import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

export const RocketApi = createAsyncThunk('rocket/rocketapi', async () => {
  const res = await fetch('https://api.spacexdata.com/v4/rockets');
  const data = res.json();
  return data;
});

const initialState = {
  rockets: [],
};

const rockets = createSlice({
  name: 'rocket',
  initialState,
  reducers: {
    reserve: (state, action) => ({
      rockets: state.rockets.map(
        (rock) => (rock.id !== action.payload ? rock : { ...rock, reserve: true }),
      ),
    }),
    cancel: (state, action) => ({
      rockets: state.rockets.map(
        (rock) => (rock.id !== action.payload ? rock : { ...rock, reserve: false }),
      ),
    }),
  },
  extraReducers: (builder) => {
    builder.addCase(RocketApi.fulfilled, (state, action) => {
      const show = action.payload;
      return {
        rockets: show.map((rocket) => (
          {
            id: rocket.id,
            rocket_name: rocket.name,
            images: rocket.flickr_images,
            description: rocket.description,
            link: rocket.wikipedia,
            reserve: false,
          }
        )),
      };
    });
  },
});

export default rockets.reducer;
export const { reserve, cancel } = rockets.actions;
