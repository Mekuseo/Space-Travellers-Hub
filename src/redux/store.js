import { configureStore } from '@reduxjs/toolkit';
import rocketSlice from './Rockets/RocketSlice';

const store = configureStore({
  reducer: {
    Rocket: rocketSlice,
  },
});

export default store;
