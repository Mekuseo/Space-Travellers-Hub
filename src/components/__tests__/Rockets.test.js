import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Rockets from '../Rockets';

test('home component match snapshot', () => {
  const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON;
  expect(tree).toMatchSnapshot();
});
