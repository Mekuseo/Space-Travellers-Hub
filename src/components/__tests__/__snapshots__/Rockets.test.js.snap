// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component match snapshot 1`] = `[Function]`;
